api/app/auth.py

```diff
@@ -55,3 +55,18 @@ async def authentication_required(token=Depends(authenticate)):
             headers={'WWW-Authenticate': 'Bearer'}
         )
     return token
+
+
+async def auth_with_set_fire_starts_permission_required(token=Depends(authentication_required)):
+    """ Only return requests that have set fire starts permission """
+    try:
+        roles = token.get('resource_access', None).get('wps-web', None).get('roles', None)
+        if 'hfi_set_fire_starts' not in roles:
+            raise HTTPException
```

Comment:
I think we should cover this edge case in our unit tests somewhere (request to change fire starts without permission).